### ISSUE TITLE
[BUG] Microphone not released after adding task by voice.

### ISSUE THREAD

**Issue by tiptruck** [TechbeeAT/jtxBoard#473]
**Describe the bug**
Adding a task using the speech to text icon.  The text is entered correctly but then the microphone does not switch off.
  * The green microphone is use indicator on the task bar is on

The time it takes for the green indicator to go out varies from about 10s to 1min.  The microphone is released if I stop JTX.  During this period, it is not possible to start recording a new item.

It looks like it is trying to do 'continued conversation' like in Google Assistant, so I disabled 'continued conversation' in Google Assistant, but that made no difference.

**To Reproduce**
1. Press the quick add item icon on the bottom left of the JTX Board
2. Press the microphone icon
   * icon changes so there is no \ through it
3. Record the task summary and description
4. Click microphone to disable
  * The \ returns in the icon briefly then disappears and recording resumes.  
  *  Clicking Save and Close causes the window to close but not to disable the microphone.  
  * The green microphone in use indicator remains on for a random amount of time or until JTX is stopped.

**Expected Behaviour**
New task created and microphone released and ready to be used again.

**Screenshots**

**Device and version:**
 - Device: [e.g. Samsung Galaxy S22] Samsung Tab S7 and Samsung A71-5G
 - Android version [e.g. Android 11] 13
 - jtx Board version: [e.g. 2.1.0] 2.03.01-rc04.ose (203010004)
 - Downloaded from: [e.g. Google Play, f-Droid, ...] F-Droid

**Comment by patrickunterwegs** [comment]
Hey @tiptruck , 
thanks, but let me first describe the logic: 
When you start the speech to text recognition, it's up to the system how long it will listen. Some time ago there was another bug ticket that the speech to text recognition stopped after some time and you needed to push the button again - which turned out a bit cumbersome. So I changed to logic a bit: When the system is listening and you say something, it will be processed. When the listening stops (automatically), the text is put into the text field and the system explicitely starts listening again. Then either the same process starts again unless no speech was recognised, in this case listening will stop.

Basically I think the logic is okay. As it's up to the system how long the speech to text engine continues listening, I can't really say when the microphone access stops. 

But as a compromise/fix I would suggest that in any case the speech recognition should stop (be destroyed) when you close the dialog. What do you think?

**Comment by patrickunterwegs** [comment]
From your description I also see that you click on the speech to text icon again to actually stop it. 
This is not implemented like that. If you click again, the speech recognizer would start listening again. 

But I get that this is confusing. I'll check that and make the speech recognition stop explicitely when you click of the icon while it's listening.

**Comment by tiptruck** [comment]
Dear @patrickunterwegs,

Thanks for explaining the logic.  I think that the icon is not representing the state of the system correctly.

Closing the window to stop recognition makes sense in any case. 

Regards Craig

**Comment by patrickunterwegs** [comment]
Hey Craig, 

yes, I have fixed the behavior that the recognition explicitely stops when the dialog is closed.
I have also improved the microphone button behavior to explicitely stop listening when clicking again and also to react immediately when you click the first time. (Before it changed when the system was really listening which could be delayed).

Regrettably I couldn't find anything about the microphone access indicator. The improvements will make it disappear faster, but the green dot still stays 2-3 seconds after stopping, even when the speech recognition object was destroyed... I think there's nothing I can do about this, I believe this is just the behavior of Android and/or the speech reconition service. But I still hope this improvement will be satisfactory 😉 

Anyway, if you check it and still find room for improvement, we can always discuss!
Cheers, 
Patrick

**Comment by tiptruck** [comment]
Thanks Patrick. It sounds great and I'll look forward to further discussions. :)

Regards Craig

--
Craig Tipping,
When you stop learning, stop listening, stop looking and asking questions, always new questions, then it is time to die. - Lillian Eugenia Smith

-------- Original Message --------
